src/components/verification-grid/verification-grid.ts

```diff
@@ -933,6 +1054,158 @@ export class VerificationGridComponent extends WithShoelace(AbstractComponent(Li
   }
 
+  /**
+   * A common method that can be used to create consistent selection behavior
+   * across the different selection methods (click, alt, arrow, tab & highlight)
+   */
+  private processSelection(
+    tileIndices: number | number[],
+    { additive = false, toggle = false, range = false, focus = false }: SelectionOptions = {},
+  ): void {
+    if (!this.canSubSelect()) {
+      return;
+    }
+
+    this.isCurrentAutoSelected = false;
+
+    let selectionBehavior = this.selectionBehavior;
+    if (selectionBehavior === "default") {
+      selectionBehavior = this.isMobileDevice() ? "tablet" : "desktop";
+    }
+
+    const selectionIndex = tileIndices;
+
+    // in desktop mode, unless the ctrl key is held down, clicking an element
+    // removes all other selected items
+    // while it is not possible to press the ctrl key on a tablet, the user can
+    // still overwrite the selection behavior using the selection-behavior
+    // attribute. Therefore, we have to check that we are not on explicitly
+    // using tablet selection mode.
+    if (selectionBehavior === "desktop" && !toggle && !additive && !focus) {
+      this.removeSubSelection();
+    }
+
+    const iterableIndices = Array.isArray(selectionIndex) ? selectionIndex : [selectionIndex];
+    for (const tileIndex of iterableIndices) {
+      // If the "focus" selection behavior is set, we want to focus the tile but
+      // not select it.
+      if (focus) {
+        this.focusTile(tileIndex);
+        this.focusHead = tileIndex;
+      } else if (range) {
+        // if the user has never selected an item before, the multiSelectHead will be "null"
+        // in this case, we want to start selecting from the clicked tile
+        this.rangeSelectionHead ??= tileIndex;
+        const selectionTail = tileIndex;
+
+        this.addSubSelectionRange(this.rangeSelectionHead, selectionTail);
+      } else if (additive) {
+        this.selectTile(tileIndex);
+        this.rangeSelectionHead = tileIndex;
+      } else {
+        // if we reach this point, we know that the user is not performing a
+        // range selection because range selection performs an early return
+        this.toggleTileSelection(tileIndex);
+        this.rangeSelectionHead = tileIndex;
+      }
+    }
+
+    this.updateSubSelection();
+  }
+
+  private resetSelectionHead(): void {
+    this.rangeSelectionHead = null;
+  }
+
+  private updateSelectionHead(value: number | null, options?: SelectionOptions): void {
+    const refinedOptions: SelectionOptions = {
+      focus: options?.toggle,
+      range: options?.range,
+    };
+
+    if (value !== null) {
+      this.processSelection(value, refinedOptions);
+    } else {
+      this.removeSubSelection();
+    }
+  }
+
+  private selectionHeadUndecided(selectionIndex: number): void {
+    let firstUndecidedTile: number | null = null;
+    const nextUndecidedTile = Array.from(this.gridTiles).findIndex((tile, index) => {
```

Comment:
maybe a comment to the effect of: "find both the first undecided tile, and the next undecided tile after the current selection. first undecided will be different if the user has undecided tiles before the current selection"

Reply:
<img width="925" height="589" alt="image" src="https://github.com/user-attachments/assets/d193362c-22dc-4075-9385-06c7227d83f4" />